### ISSUE TITLE
Allow automatic reports of mails moved to the spam folder

### ISSUE THREAD

**Issue by vaf-hub** [tutao/tutanota#3015]
Currently, when users move a mail to the spam folder the mail is not reported as spam to the server.

What we should do:
When a user moves a mail to the spam folder for the first time (after this change), we should open a dialog and ask the user if from now on they a) automatically want to report each mail moved to the server as spam in cleartext, b) be asked about this every time or c) never report mails moved to spam. There will be an option in the mail settings to change this configuration.

We will not allow "no spam" reports to avoid abuse and don't get a conflict with privacy by offering an option to send valid emails to us as plain text automatically.

~~The selected setting should also apply for mails moved based on inbox rules.~~

Attention: The solution to this issue contains a model change!

**Comment by vaf-hub** [comment]
After a discussion with @armhub we decided that we also want to allow users to automatically report mails moved out of the spam folder, since those are the ones that were falsely classified as spam.

**Comment by vaf-hub** [comment]
Discussed with @charlag that we should debounce the actual report of mails moved and show some SnackBar with a button (which is yet to create) to undo the move before the report request is made. because otherwise we would also report mails that were accidentally moved to the spam folder.

**Comment by armhub** [comment]
Test notes
- [ ] Move email to spam, confirm reporting and do not remember decision, move another email to spam -> you are asked again to confirm
- [ ] Move email to spam, confirm reporting remember decision, move another email to spam -> you are not asked again and the email is reported
- [ ] Move email to spam, do not report, remember decision, move another email to spam -> you are not asked again and email is not reported
- [ ] Change the decision in the email settings and make sure it works as selected
- [ ] Regression test reporting email as phishing and spam in the email more menu

In each test case check the network requests to find out if the email is reported to the server.

**Comment by johnbotris** [comment]
some thoughts

* The snack bar button says "undo", but it only cancels reporting the mails. I somehow expect the button to undo the mail being moved to spam. Maybe it should say cancel?
* If i move multiple mails to spam but one at a time, then the snackbars for each operation will show individually, and only after the timeout from the previous, it's a bit annoying. if we buffer the report operations and extend the life of the snackbar for each mail moved to spam, i think that could work better. Maybe we don't change it for the release but I would open an issue

**Comment by vaf-hub** [comment]
We also have to retest this issue because some changes from the review were not included. I changed the button label to "cancel" now. and will create a new issue for your second suggestion.
